packages/data-explorer/src/views/MainView.vue

```diff
@@ -143,24 +146,30 @@ export default Vue.extend({
         this.setTableName(tableName)
       }
       if (this.dataDisplayLayout === 'CardView') {
-        this.fetchCardViewData()
+        res = await this.fetchCardViewData(this.pagination)
       } else {
-        this.fetchTableViewData()
+        res = await this.fetchTableViewData(this.pagination)
       }
       this.loading = false
+      return res
+    },
+    async paginator () {
+      const res = await this.fetchViewData(this.$route.params.entity)
+      return { count: res.response.page.totalElements }
```

Comment:
would suggest to move this into the store, then there would be no need to move around the 'res' on the outside

Reply:
Moved fetchCardViewData to actions